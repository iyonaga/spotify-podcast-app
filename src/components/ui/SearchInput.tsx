import SearchIcon from '../icons/Search';

interface Props {
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  handleKeyPress: (e: React.KeyboardEvent<HTMLInputElement>) => void;
  searchInput: string;
}

const SearchInput: React.VFC<Props> = ({
  handleChange,
  handleKeyPress,
  searchInput,
}) => {
  return (
    <div className="group relative">
      <div className="absolute top-[15px] left-[23px]">
        <SearchIcon className="fill-gray group-focus-within:fill-white" />
      </div>
      <input
        className="px-[55px] w-full h-[50px] text-white bg-transparent rounded-[25px] border-2 border-gray focus:border-white border-solid outline-none"
        type="text"
        placeholder="Search"
        onChange={handleChange}
        onKeyPress={handleKeyPress}
        value={searchInput}
      />
    </div>
  );
};

export default SearchInput;
