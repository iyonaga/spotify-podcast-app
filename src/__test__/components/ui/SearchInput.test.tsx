import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchInput from '@/components/ui/SearchInput';

const handleChange = jest.fn();
const handleKeyPress = jest.fn();
const query = '';
const props = {
  handleChange,
  handleKeyPress,
  query,
};

describe('components/ui/SearchInput.tsx', () => {
  beforeEach(() => {
    handleChange.mockClear();
    handleKeyPress.mockClear();
  });

  test('スナップショットが一致する', () => {
    const { asFragment } = render(<SearchInput {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('valueが設定される', () => {
    const { rerender } = render(<SearchInput {...props} />);
    expect(screen.getByRole('textbox')).toHaveValue(query);
    rerender(<SearchInput {...props} query="dummy" />);
    expect(screen.getByRole('textbox')).toHaveValue('dummy');
  });

  test('onChangeイベントが発火する', () => {
    render(<SearchInput {...props} />);
    const input = screen.getByRole('textbox');
    userEvent.type(input, 'test');
    expect(input).toHaveValue('test');
    expect(handleChange).toHaveBeenCalledTimes(4);
  });

  test('onKeyPressイベントが発火する', () => {
    render(<SearchInput {...props} />);
    const input = screen.getByRole('textbox');
    fireEvent.keyPress(input, { key: 'Enter', keyCode: 13 });
    expect(handleKeyPress).toHaveBeenCalledTimes(1);
  });
});
