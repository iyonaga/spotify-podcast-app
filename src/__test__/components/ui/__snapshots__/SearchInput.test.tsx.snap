// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/SearchInput.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <div
    class="group relative"
  >
    <div
      class="absolute top-[15px] left-[23px]"
    >
      <svg
        class="fill-gray group-focus-within:fill-white"
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.0303 13.9697C14.7374 13.6768 14.2626 13.6768 13.9697 13.9697C13.6768 14.2626 13.6768 14.7374 13.9697 15.0303L15.0303 13.9697ZM18.4697 19.5303C18.7626 19.8232 19.2374 19.8232 19.5303 19.5303C19.8232 19.2374 19.8232 18.7626 19.5303 18.4697L18.4697 19.5303ZM14.1764 14.1619L13.6454 13.6322L13.6453 13.6322L14.1764 14.1619ZM13.9697 15.0303L18.4697 19.5303L19.5303 18.4697L15.0303 13.9697L13.9697 15.0303ZM0.25 8.71429C0.25 13.389 4.03959 17.1786 8.71429 17.1786V15.6786C4.86802 15.6786 1.75 12.5605 1.75 8.71429H0.25ZM8.71429 17.1786C11.0554 17.1786 13.1759 16.2271 14.7074 14.6915L13.6453 13.6322C12.3838 14.8972 10.6409 15.6786 8.71429 15.6786V17.1786ZM14.7074 14.6915C16.2337 13.1612 17.1786 11.0474 17.1786 8.71429H15.6786C15.6786 10.6343 14.9026 12.3717 13.6454 13.6322L14.7074 14.6915ZM17.1786 8.71429C17.1786 4.03959 13.389 0.25 8.71429 0.25V1.75C12.5605 1.75 15.6786 4.86802 15.6786 8.71429H17.1786ZM8.71429 0.25C4.03959 0.25 0.25 4.03959 0.25 8.71429H1.75C1.75 4.86802 4.86802 1.75 8.71429 1.75V0.25Z"
        />
      </svg>
    </div>
    <input
      class="px-[55px] w-full h-[50px] text-white bg-transparent rounded-[25px] border-2 border-gray focus:border-white border-solid outline-none"
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
